provider/testdata/src/lib.rs

```diff
@@ -84,32 +84,20 @@
 
 extern crate alloc;
 
-#[path = "../metadata.rs.data"]
+#[path = "../data/metadata.rs.data"]
 mod metadata;
 
 pub mod versions {
     //! Functions to access version info of the ICU test data.
 
     /// Gets the CLDR tag used as the test data source (for formatters, likely subtags, ...)
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// assert_eq!("42.0.0", icu_testdata::versions::cldr_tag());
```

Comment:
Because now that the tag is set by the generated metadata, I don't want a doc test to assert a particular value.

Reply:
Nit: I prefer to keep this because every public function should be covered by a docs test; we already have had discussion that it's okay to have localized strings in docs tests, which is no different from tag numbers.

Reply:
This is a constant. You don't test a constant with itself

Reply:
It's not a constant? It's a public function that returns stuff. We want to make sure the function works.

Reply:
We can test that it "returns stuff", but I don't want to test that it returns an exact version number, because that's just duplicating constants and giving clients wrong impressions (i.e. that this is somewhat stable).

We can add
```rust
#[test]
fn version_returns_stuff() {
  crate::versions::cldr()
}
```
to see that it doesn't panic, but I think that's overkill.

Reply:
Could we add a dev-dependency on icu_datagen and assert that the string equals the LATEST_TESTED tag string?

I really do want to make sure this function is tested in CI, since we have a client using it, and it has caused problems before

Reply:
What problems has it caused?

Sure we could add that dependency, but that test would be pretty circular. `ci-job-testdata` already asserts that the values in `metadata.rs.data` are the same as `LATEST_TESTED`.

Reply:
I still strongly prefer to test the tag string. I think it has value to docs readers because they can easily see what CLDR version all of our docs tests are using.

Reply:
These are allocating functions because they have given us the flexibility to change the underlying storage.

Reply:
Yes and it's taken the flexibility to make these constants. I'll revert this.